src/MoBi.Presentation/UICommand/RefreshSimulationSettingsUICommand.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Linq;
+using MoBi.Core.Domain.Model;
+using MoBi.Presentation.Tasks;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Services;
+using OSPSuite.Presentation.UICommands;
+
+namespace MoBi.Presentation.UICommand
+{
+   public class RefreshSimulationSettingsUICommand : ActiveObjectUICommand<SimulationSettings>
+   {
+      private readonly ISimulationUpdateTask _simulationUpdateTask;
+      private readonly IMoBiContext _context;
+
+      public RefreshSimulationSettingsUICommand(
+         ISimulationUpdateTask simulationUpdateTask,
+         IMoBiContext context,
+         IActiveSubjectRetriever activeSubjectRetriever) :
+         base(activeSubjectRetriever)
+      {
+         _simulationUpdateTask = simulationUpdateTask;
+         _context = context;
+      }
+
+      protected override void PerformExecute()
+      {
+         var simulation = _context.CurrentProject.Simulations.FirstOrDefault(x => x.Settings.Equals(Subject));
```

Comment:
meh. cannopt the command be for a Simulation instead? That's a bit nasty

Reply:
I created an object to use as the ```TreeNode.Tag``` that holds the Simulation instead of the SimulationSettings so that this command can be used directly on the Simulation.

Reply:
I might not need the viewitem now since the DTO is a viewitem